tests/test-lapp.lua

```diff
@@ -147,6 +147,19 @@ check (false_flag,{'-g','-f'},{f=false,g=true})
 check (false_flag,{'-g','--'},{f=true,g=true})
 check (false_flag,{'-g','--','-a','frodo'},{f=true,g=true; '-a','frodo'})
 
+-- check that negative number arugments are accepted
+local negative_argument = [[
+    -f, --flag      short flag
```

Comment:
should we add test case for;
```lua
    -2, --flag2       bad flag that shouldn't be allowed   --> should throw an error when trying to use
```

Reply:
and that begs the question whether a long-flag is allowed to start with a number?

Reply:
Ok, so python argparse also treat negative numbers as flags.  So I think it is best to keep the default behavior as it is now. I will just modify the penlight code for my use case.

Reply:
Other BSD shell commands like `ls` also treat `-2` as a flag option. So it seems to be standard, just not well documented anywhere I can find 😅.